core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -46,6 +46,7 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
 
```

Comment:
The additions to this file seem overly complex for just grabbing port resources.  Maybe all this complexity is necessary, but I'm skeptical.  I started in on it and got lost.  I don't have time to review it all today.  Please ensure there is no more room for simplification.